lib/kafka/consumer.rb

```diff
@@ -100,12 +100,16 @@ def stop
     # idea to simply pause the partition until the error can be resolved, allowing
     # the rest of the partitions to continue being processed.
     #
+    # If the `timeout` argument is passed, the partition will automatically be
+    # resumed when the timeout expires.
+    #
     # @param topic [String]
     # @param partition [Integer]
+    # @param timeout [Integer] the number of seconds to pause the partition for.
     # @return [nil]
-    def pause(topic, partition)
-      @paused_partitions[topic] ||= Set.new
-      @paused_partitions[topic] << partition
+    def pause(topic, partition, timeout: nil)
+      @paused_partitions[topic] ||= {}
+      @paused_partitions[topic][partition] = timeout && timeout + Time.now
```

Comment:
`TypeError: Time can't be coerced into Fixnum`

I think `Time.now + timeout` will work better.